cpp/include/raft/linalg/unary_op.cuh

```diff
@@ -71,6 +73,79 @@ void writeOnlyUnaryOp(OutType* out, IdxType len, Lambda op, cudaStream_t stream)
   detail::writeOnlyUnaryOpCaller(out, len, op, stream);
 }
 
+/**
+ * @defgroup unary_op Element-Wise Unary Operations
+ * @{
+ */
+
+/**
+ * @brief perform element-wise binary operation on the input arrays
+ * @tparam InType Input Type raft::device_mdspan
+ * @tparam Lambda the device-lambda performing the actual operation
+ * @tparam OutType Output Type raft::device_mdspan
+ * @tparam TPB threads-per-block in the final kernel launched
+ * @param[in] handle raft::handle_t
+ * @param[in] in Input
+ * @param[out] out Output
+ * @param[in] op the device-lambda
+ * @note Lambda must be a functor with the following signature:
+ *       `InType func(const InType& val);`
+ */
+template <typename InType,
+          typename Lambda,
+          typename OutType,
+          int TPB  = 256,
```

Comment:
This is exposing underlying implementation details and we should remove it from public APIs everywhere. We should also do a quick grep through cuml and cugraph to make sure this isn't being set to anything other than 256.